### PR TITLE
Observer newspaper page

### PR DESCRIPTION
**Pull request by jennysivapalan**
Following #11094 and #11121 this adds `/theobserver` and `/theobserver/[year][month]/[day]` newspaper book pages. 
- `/theobserver` will show the latest Sunday edition of The Observer based on the tag `theobserver/news`.
- `/theobserver/[year][month]/[day]` will show the Observer content for that date. If the date returns zero results (e.g. not a Sunday or a date in the future) then it will redirect to `/theobserver`. `/theguardian/[year][month]/[day]` rules have been updated to follow suit but to redirect to `/theguardian`. This follows the pattern of R2.
- `/theobserver/[year][month]/[day]/all` will redirect to `/theobserver/[year][month]/[day]` 

The section navigation for this page may need updating but it should be a minor tweak and can go out as a separate PR.

screen shot 2015-11-19 at 16 52 19

### CODE REVIEW

**Comment by mattandrews** [comment]
:eyeglasses:

File: applications/app/services/NewspaperQuery.scala [hunk 1]
```diff
@@ -124,4 +130,11 @@ object NewspaperQuery extends ExecutionContexts with Dates with Logging {
   private def getNewspaperPageNumber(content: ApiContent) = content.fields.getOrElse(Map.empty).get("newspaperPageNumber").map(_.toInt)
 
   def lowercaseDisplayName(s: String) = if(s.equals("UK news") || s.equals("US news")) s else s.toLowerCase()
+
+  def getPastSundayDateFor(date: DateTime) = {
+    val daysSinceSunday = DateTimeConstants.SUNDAY - date.getDayOfWeek - 7
+    date.minusDays(Math.abs(daysSinceSunday))
```
Comment (JustinPinner) [inline applications/app/services/NewspaperQuery.scala]: If today is Sunday, will this get last Sunday's content (7 - 7) - 7 = -7?

File: applications/test/services/NewspapeQueryTest.scala [hunk 2]
```diff
@@ -1,9 +1,10 @@
 package services
```
Comment (JustinPinner) [inline applications/test/services/NewspapeQueryTest.scala]: I spy a typo in the file and class name (NewspapeQueryTest.scala) :confused:

File: applications/app/services/NewspaperQuery.scala [hunk 3]
```diff
@@ -22,26 +22,32 @@ object NewspaperQuery extends ExecutionContexts with Dates with Logging {
...
 
-  def fetchTodaysPaper: Future[List[FaciaContainer]] =
-    bookSectionContainers(DateTime.now(DateTimeZone.UTC))
+  def fetchLatestGuardianNewspaper: Future[List[FaciaContainer]] = {
+    val date = DateTime.now(DateTimeZone.UTC)
+    bookSectionContainers("theguardian/mainsection", getLatestGuardianPageFor(date))
+  }
 
+  def fetchLatestObserverNewspaper: Future[List[FaciaContainer]] = {
+    val now = DateTime.now(DateTimeZone.UTC)
```
Comment (JustinPinner) [inline applications/app/services/NewspaperQuery.scala]: Teeny niggle - might we use `val date = ...` here or `val now = ...` on line 28?

File: applications/app/services/NewspaperQuery.scala (hunk 1 above)
Comment (jennysivapalan) [inline applications/app/services/NewspaperQuery.scala]: Good spot :eyes:

**Comment by jennysivapalan** [comment]
@JustinPinner Thanks have made those corrections
@mattandrews :newspaper:

**Comment by JustinPinner** [comment]
:+1: :sheep: :it:

**Pull request merged**